drivers/infiniband/core/uverbs_std_types.c

```diff
@@ -574,6 +574,78 @@ int uverbs_reg_mr_handler(struct ib_device *ib_dev,
 }
 
 DECLARE_UVERBS_ATTR_SPEC(
+	uverbs_rereg_mr_spec,
+	UVERBS_ATTR_IDR(REREG_MR_HANDLE, UVERBS_TYPE_MR, UVERBS_ACCESS_READ,
+			UA_FLAGS(UVERBS_ATTR_SPEC_F_MANDATORY)),
+	UVERBS_ATTR_IDR(REREG_MR_PD_HANDLE, UVERBS_TYPE_PD, UVERBS_ACCESS_READ,
+			UA_FLAGS(UVERBS_ATTR_SPEC_F_MANDATORY)),
+	UVERBS_ATTR_PTR_IN(REREG_MR_CMD, struct ib_uverbs_rereg_mr,
```

Comment:
I think it's better to break the struct to attr fields.
Meaning, having an attribute for "change access", another one for "translation", etc.